[BUG] AAS with Administration.TemplateId created but not registered

**Describe the bug**
I am trying to add AAS with Administration elements containing a TemplateId to my BaSyx environment, which won't work as expected. I broke the problem down by minimalizing the content of my AAS to modelType and id. Posting 

```JSON
{
  "modelType": "AssetAdministrationShell",
  "id": "https://www.GALAB.th-koeln.de/aut/aas/1731332658219"
}
```

works as expected. The AAS is created (response 201), as well as the corrosponding AasDescriptor. Posting 

```JSON
{
  "modelType": "AssetAdministrationShell",
  "id": "https://www.GALAB.th-koeln.de/aut/aas/1731332658219",
  "administration": {
    "templateId": "sensor"
  }
}
```

gives response 500 _Internal Server Error_. The logs (added below) of the BaSyx Environment container show, that there was an error while creating the AasDescriptor because some string constraints are not met. According to the swagger-hub docs the templateId is of type _string_. The current [AAS-Spec](https://industrialdigitaltwin.org/wp-content/uploads/2024/06/IDTA-01001-3-0-1_SpecificationAssetAdministrationShell_Part1_Metamodel.pdf#page=96) says it is of type _Identifier_ which is specified as a string with a length between 1 and 2000, no further restrictions.
Nevertheless and contradict to the response message the AAS is created in the Environment, but not added to the Registry.

**To Reproduce**
Steps to reproduce the behavior:
1. Set up Environment and Registry
2. Post AAS containing Administration with TemplateId
3. See response and logs of Environment

**Expected behavior**
Creation of AAS and ShellDescriptor, response 201.

**BaSyx:**
- BaSyx Version v2.0.0.SNAPSHOT 
- AAS Version DotAAS V3

**LOGS**
```
2024-11-13 10:30:19 2024-11-13T09:30:19.960Z ERROR 1 --- [AAS Environment] [nio-8081-exec-5] o.a.c.c.C.[.[.[/].[dispatcherServlet]    : Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed: org.eclipse.digitaltwin.basyx.core.exceptions.RepositoryRegistryLinkException: Unable to link the element with id 'https://www.GALAB.th-koeln.de/aut/aas/1731332658219' with the Registry.] with root cause
2024-11-13 10:30:19 
2024-11-13 10:30:19 org.eclipse.digitaltwin.basyx.aasregistry.client.ApiException: postAssetAdministrationShellDescriptor call failed with: 400 - {"messages":[{"code":"Bad Request","correlationId":null,"messageType":"Exception","text":"Field error in object 'assetAdministrationShellDescriptor' on field 'administration.templateId': rejected value [sensor]; codes [Pattern.assetAdministrationShellDescriptor.administration.templateId,Pattern.administration.templateId,Pattern.templateId,Pattern.java.lang.String,Pattern]; arguments [org.springframework.context.support.DefaultMessageSourceResolvable: codes [assetAdministrationShellDescriptor.administration.templateId,administration.templateId]; arguments []; default message [administration.templateId],[Ljakarta.validation.constraints.Pattern$Flag;@774146d6,^([\\t\\n\\r \\ud7ff\\ue000-\\ufffd]|\\ud800[\\udc00-\\udfff]|[\\ud801-\\udbfe][\\udc00-\\udfff]|\\udbff[\\udc00-\\udfff])*$]; default message [must match \"^([\\t\\n\\r \\ud7ff\\ue000-\\ufffd]|\\ud800[\\udc00-\\udfff]|[\\ud801-\\udbfe][\\udc00-\\udfff]|\\udbff[\\udc00-\\udfff])*$\"]","timestamp":1731490219.956723480}]}
2024-11-13 10:30:19     at org.eclipse.digitaltwin.basyx.aasregistry.client.api.RegistryAndDiscoveryInterfaceApi.getApiException(RegistryAndDiscoveryInterfaceApi.java:125) ~[basyx.aasregistry-client-native-2.0.0-milestone-04.jar!/:2.0.0-milestone-04]
2024-11-13 10:30:19     at org.eclipse.digitaltwin.basyx.aasregistry.client.api.RegistryAndDiscoveryInterfaceApi.postAssetAdministrationShellDescriptorWithHttpInfo(RegistryAndDiscoveryInterfaceApi.java:833) ~[basyx.aasregistry-client-native-2.0.0-milestone-04.jar!/:2.0.0-milestone-04]
2024-11-13 10:30:19     at org.eclipse.digitaltwin.basyx.aasregistry.client.api.RegistryAndDiscoveryInterfaceApi.postAssetAdministrationShellDescriptor(RegistryAndDiscoveryInterfaceApi.java:811) ~[basyx.aasregistry-client-native-2.0.0-milestone-04.jar!/:2.0.0-milestone-04]
2024-11-13 10:30:19     at org.eclipse.digitaltwin.basyx.aasrepository.feature.registry.integration.RegistryIntegrationAasRepository.integrateAasWithRegistry(RegistryIntegrationAasRepository.java:134) ~[basyx.aasrepository-feature-registry-integration-2.0.0-milestone-04.jar!/:2.0.0-milestone-04]
2024-11-13 10:30:19     at org.eclipse.digitaltwin.basyx.aasrepository.feature.registry.integration.RegistryIntegrationAasRepository.createAas(RegistryIntegrationAasRepository.java:83) ~[basyx.aasrepository-feature-registry-integration-2.0.0-milestone-04.jar!/:2.0.0-milestone-04]
2024-11-13 10:30:19     at org.eclipse.digitaltwin.basyx.aasrepository.http.AasRepositoryApiHTTPController.postAssetAdministrationShell(AasRepositoryApiHTTPController.java:92) ~[basyx.aasrepository-http-2.0.0-milestone-04.jar!/:2.0.0-milestone-04]
2024-11-13 10:30:19     at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
2024-11-13 10:30:19     at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[na:na]
2024-11-13 10:30:19     at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
2024-11-13 10:30:19     at java.base/java.lang.reflect.Method.invoke(Method.java:569) ~[na:na]
2024-11-13 10:30:19     at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:354) ~[spring-aop-6.1.11.jar!/:6.1.11]
2024-11-13 10:30:19     at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:196) ~[spring-aop-6.1.11.jar!/:6.1.11]
2024-11-13 10:30:19     at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163) ~[spring-aop-6.1.11.jar!/:6.1.11]
2024-11-13 10:30:19     at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:768) ~[spring-aop-6.1.11.jar!/:6.1.11]
2024-11-13 10:30:19     at org.springframework.validation.beanvalidation.MethodValidationInterceptor.invoke(MethodValidationInterceptor.java:174) ~[spring-context-6.1.11.jar!/:6.1.11]
2024-11-13 10:30:19     at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184) ~[spring-aop-6.1.11.jar!/:6.1.11]
2024-11-13 10:30:19     at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:768) ~[spring-aop-6.1.11.jar!/:6.1.11]
2024-11-13 10:30:19     at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:720) ~[spring-aop-6.1.11.jar!/:6.1.11]
2024-11-13 10:30:19     at org.eclipse.digitaltwin.basyx.aasrepository.http.AasRepositoryApiHTTPController$$SpringCGLIB$$0.postAssetAdministrationShell(<generated>) ~[basyx.aasrepository-http-2.0.0-milestone-04.jar!/:2.0.0-milestone-04]
2024-11-13 10:30:19     at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
2024-11-13 10:30:19     at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[na:na]
2024-11-13 10:30:19     at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
2024-11-13 10:30:19     at java.base/java.lang.reflect.Method.invoke(Method.java:569) ~[na:na]
2024-11-13 10:30:19     at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:255) ~[spring-web-6.1.11.jar!/:6.1.11]
2024-11-13 10:30:19     at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:188) ~[spring-web-6.1.11.jar!/:6.1.11]
2024-11-13 10:30:19     at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:118) ~[spring-webmvc-6.1.11.jar!/:6.1.11]
2024-11-13 10:30:19     at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:926) ~[spring-webmvc-6.1.11.jar!/:6.1.11]
2024-11-13 10:30:19     at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:831) ~[spring-webmvc-6.1.11.jar!/:6.1.11]
2024-11-13 10:30:19     at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87) ~[spring-webmvc-6.1.11.jar!/:6.1.11]
2024-11-13 10:30:19     at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1089) ~[spring-webmvc-6.1.11.jar!/:6.1.11]
2024-11-13 10:30:19     at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:979) ~[spring-webmvc-6.1.11.jar!/:6.1.11]
2024-11-13 10:30:19     at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1014) ~[spring-webmvc-6.1.11.jar!/:6.1.11]
2024-11-13 10:30:19     at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:914) ~[spring-webmvc-6.1.11.jar!/:6.1.11]
2024-11-13 10:30:19     at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:590) ~[tomcat-embed-core-10.1.26.jar!/:na]
2024-11-13 10:30:19     at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:885) ~[spring-webmvc-6.1.11.jar!/:6.1.11]
2024-11-13 10:30:19     at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:658) ~[tomcat-embed-core-10.1.26.jar!/:na]
2024-11-13 10:30:19     at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:195) ~[tomcat-embed-core-10.1.26.jar!/:na]
2024-11-13 10:30:19     at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140) ~[tomcat-embed-core-10.1.26.jar!/:na]
2024-11-13 10:30:19     at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51) ~[tomcat-embed-websocket-10.1.26.jar!/:na]
2024-11-13 10:30:19     at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164) ~[tomcat-embed-core-10.1.26.jar!/:na]
2024-11-13 10:30:19     at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140) ~[tomcat-embed-core-10.1.26.jar!/:na]
2024-11-13 10:30:19     at org.eclipse.digitaltwin.basyx.http.HttpBaSyxHeader.doFilterInternal(HttpBaSyxHeader.java:53) ~[basyx.http-2.0.0-milestone-04.jar!/:2.0.0-milestone-04]
2024-11-13 10:30:19     at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.1.11.jar!/:6.1.11]
2024-11-13 10:30:19     at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164) ~[tomcat-embed-core-10.1.26.jar!/:na]
2024-11-13 10:30:19     at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140) ~[tomcat-embed-core-10.1.26.jar!/:na]
2024-11-13 10:30:19     at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100) ~[spring-web-6.1.11.jar!/:6.1.11]
2024-11-13 10:30:19     at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.1.11.jar!/:6.1.11]
2024-11-13 10:30:19     at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164) ~[tomcat-embed-core-10.1.26.jar!/:na]
2024-11-13 10:30:19     at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140) ~[tomcat-embed-core-10.1.26.jar!/:na]
2024-11-13 10:30:19     at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93) ~[spring-web-6.1.11.jar!/:6.1.11]
2024-11-13 10:30:19     at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.1.11.jar!/:6.1.11]
2024-11-13 10:30:19     at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164) ~[tomcat-embed-core-10.1.26.jar!/:na]
2024-11-13 10:30:19     at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140) ~[tomcat-embed-core-10.1.26.jar!/:na]
2024-11-13 10:30:19     at org.springframework.web.filter.ServerHttpObservationFilter.doFilterInternal(ServerHttpObservationFilter.java:113) ~[spring-web-6.1.11.jar!/:6.1.11]
2024-11-13 10:30:19     at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.1.11.jar!/:6.1.11]
2024-11-13 10:30:19     at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164) ~[tomcat-embed-core-10.1.26.jar!/:na]
2024-11-13 10:30:19     at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140) ~[tomcat-embed-core-10.1.26.jar!/:na]
2024-11-13 10:30:19     at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201) ~[spring-web-6.1.11.jar!/:6.1.11]
2024-11-13 10:30:19     at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.1.11.jar!/:6.1.11]
2024-11-13 10:30:19     at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164) ~[tomcat-embed-core-10.1.26.jar!/:na]
2024-11-13 10:30:19     at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140) ~[tomcat-embed-core-10.1.26.jar!/:na]
2024-11-13 10:30:19     at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:167) ~[tomcat-embed-core-10.1.26.jar!/:na]
2024-11-13 10:30:19     at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90) ~[tomcat-embed-core-10.1.26.jar!/:na]
2024-11-13 10:30:19     at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:483) ~[tomcat-embed-core-10.1.26.jar!/:na]
2024-11-13 10:30:19     at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:115) ~[tomcat-embed-core-10.1.26.jar!/:na]
2024-11-13 10:30:19     at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93) ~[tomcat-embed-core-10.1.26.jar!/:na]
2024-11-13 10:30:19     at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74) ~[tomcat-embed-core-10.1.26.jar!/:na]
2024-11-13 10:30:19     at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:344) ~[tomcat-embed-core-10.1.26.jar!/:na]
2024-11-13 10:30:19     at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:389) ~[tomcat-embed-core-10.1.26.jar!/:na]
2024-11-13 10:30:19     at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63) ~[tomcat-embed-core-10.1.26.jar!/:na]
2024-11-13 10:30:19     at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:904) ~[tomcat-embed-core-10.1.26.jar!/:na]
2024-11-13 10:30:19     at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1741) ~[tomcat-embed-core-10.1.26.jar!/:na]
2024-11-13 10:30:19     at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52) ~[tomcat-embed-core-10.1.26.jar!/:na]
2024-11-13 10:30:19     at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1190) ~[tomcat-embed-core-10.1.26.jar!/:na]
2024-11-13 10:30:19     at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659) ~[tomcat-embed-core-10.1.26.jar!/:na]
2024-11-13 10:30:19     at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63) ~[tomcat-embed-core-10.1.26.jar!/:na]
2024-11-13 10:30:19     at java.base/java.lang.Thread.run(Thread.java:840) ~[na:na]
```

I can confirm the bug @wasn93 described for AAS containing a `templateId` in the `administrativeInformation`.

The bug does not occur for AAS only containing `creator`, `embeddedDataSpecifications` `version` and/or `revision` in the `administrativeInformation`.

### To Reproduce
Steps to reproduce the behavior:

1. Docker compose for aas-environment, aas-registry and sm-registry; aas-environment with feature.registryintegration and basyx.environment for loading AASX on startup
2. See part of aas-environment container log

### Expected behavior
Startup without errors

### Analysis
There seems to be a problem by matching the templateId with a regex:

```
Field error in object 'assetAdministrationShellDescriptor' on field 'administration.templateId': rejected value [templateAas_id]; codes [Pattern.assetAdministrationShellDescriptor.administration.templateId,Pattern.administration.templateId,Pattern.templateId,Pattern.java.lang.String,Pattern]; arguments [org.springframework.context.support.DefaultMessageSourceResolvable: codes [assetAdministrationShellDescriptor.administration.templateId,administration.templateId]; arguments []; default message [administration.templateId],[Ljakarta.validation.constraints.Pattern$Flag;@3bce7351,^([\\t\\n\\r \\ud7ff\\ue000-\\ufffd]|\\ud800[\\udc00-\\udfff]|[\\ud801-\\udbfe][\\udc00-\\udfff]|\\udbff[\\udc00-\\udfff])*$]; default message [must match \"^([\\t\\n\\r \\ud7ff\\ue000-\\ufffd]|\\ud800[\\udc00-\\udfff]|[\\ud801-\\udbfe][\\udc00-\\udfff]|\\udbff[\\udc00-\\udfff])*$\"]
```

### BaSyx:

BaSyx Version v2.0.0.SNAPSHOT
AAS Version DotAAS V3

### Logs

```
2024-11-18 11:45:52 Loaded property file: file:application.properties
2024-11-18 11:45:52 Loaded property file: jar:nested:/application/basyxExecutable.jar/!BOOT-INF/classes/!/application.properties
2024-11-18 11:45:53   ____           _____              
2024-11-18 11:45:53  |  _ \         / ____|             
2024-11-18 11:45:53  | |_) |  __ _ | (___   _   _ __  __
2024-11-18 11:45:53  |  _ <  / _` | \___ \ | | | |\ \/ /
2024-11-18 11:45:53  | |_) || (_| | ____) || |_| | >  < 
2024-11-18 11:45:53  |____/  \__,_||_____/  \__, |/_/\_\
2024-11-18 11:45:53 ======================== __/ |======      
2024-11-18 11:45:53 AAS Environment         |___/                  
2024-11-18 11:45:53 2.0.0-PREVIEW
2024-11-18 11:45:53 
2024-11-18 11:45:53 2024-11-18T10:45:53.164Z  INFO 1 --- [AAS Environment] [           main] o.e.d.b.a.c.AasEnvironmentComponent      : Starting AasEnvironmentComponent v2.0.0-SNAPSHOT using Java 17.0.13 with PID 1 (/application/basyxExecutable.jar started by nobody in /application)
2024-11-18 11:45:53 2024-11-18T10:45:53.166Z  INFO 1 --- [AAS Environment] [           main] o.e.d.b.a.c.AasEnvironmentComponent      : No active profile set, falling back to 1 default profile: "default"
2024-11-18 11:45:54 2024-11-18T10:45:54.658Z  INFO 1 --- [AAS Environment] [           main] .s.d.r.c.RepositoryConfigurationDelegate : Bootstrapping Spring Data MongoDB repositories in DEFAULT mode.
2024-11-18 11:45:54 2024-11-18T10:45:54.671Z  INFO 1 --- [AAS Environment] [           main] .s.d.r.c.RepositoryConfigurationDelegate : Finished Spring Data repository scanning in 8 ms. Found 0 MongoDB repository interfaces.
2024-11-18 11:45:54 2024-11-18T10:45:54.730Z  INFO 1 --- [AAS Environment] [           main] faultConfiguringBeanFactoryPostProcessor : No bean named 'errorChannel' has been explicitly defined. Therefore, a default PublishSubscribeChannel will be created.
2024-11-18 11:45:54 2024-11-18T10:45:54.739Z  INFO 1 --- [AAS Environment] [           main] faultConfiguringBeanFactoryPostProcessor : No bean named 'integrationHeaderChannelRegistry' has been explicitly defined. Therefore, a default DefaultHeaderChannelRegistry will be created.
2024-11-18 11:45:55 2024-11-18T10:45:55.251Z  INFO 1 --- [AAS Environment] [           main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port 8081 (http)
2024-11-18 11:45:55 2024-11-18T10:45:55.262Z  INFO 1 --- [AAS Environment] [           main] o.apache.catalina.core.StandardService   : Starting service [Tomcat]
2024-11-18 11:45:55 2024-11-18T10:45:55.262Z  INFO 1 --- [AAS Environment] [           main] o.apache.catalina.core.StandardEngine    : Starting Servlet engine: [Apache Tomcat/10.1.31]
2024-11-18 11:45:55 2024-11-18T10:45:55.298Z  INFO 1 --- [AAS Environment] [           main] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring embedded WebApplicationContext
2024-11-18 11:45:55 2024-11-18T10:45:55.300Z  INFO 1 --- [AAS Environment] [           main] w.s.c.ServletWebServerApplicationContext : Root WebApplicationContext: initialization completed in 2074 ms
2024-11-18 11:45:55 2024-11-18T10:45:55.500Z  INFO 1 --- [AAS Environment] [           main] o.e.d.b.a.c.AasEnvironmentFeaturePrinter : -------------------- Aas Environment Features: --------------------
2024-11-18 11:45:55 2024-11-18T10:45:55.500Z  INFO 1 --- [AAS Environment] [           main] o.e.d.b.a.c.AasEnvironmentFeaturePrinter : ----------------------------------------------------------------- 
2024-11-18 11:45:56 2024-11-18T10:45:56.895Z  INFO 1 --- [AAS Environment] [           main] a.p.AasEnvironmentPreconfigurationLoader : Scanning for preconfigured AAS Environments
2024-11-18 11:45:56 2024-11-18T10:45:56.910Z  INFO 1 --- [AAS Environment] [           main] a.p.AasEnvironmentPreconfigurationLoader : Found 2 preconfigured AAS environments
2024-11-18 11:45:56 2024-11-18T10:45:56.910Z  INFO 1 --- [AAS Environment] [           main] a.p.AasEnvironmentPreconfigurationLoader : Loading AAS Environment (0/2) from file 'aas-gui.aasx'
2024-11-18 11:45:57 2024-11-18T10:45:57.634Z  INFO 1 --- [AAS Environment] [           main] a.f.r.i.RegistryIntegrationAasRepository : Shell 'templateAas_id' has been automatically linked with the Registry
2024-11-18 11:45:57 2024-11-18T10:45:57.637Z  INFO 1 --- [AAS Environment] [           main] o.e.d.b.a.base.DefaultAASEnvironment     : Could not find thumbnail resource for aas templateAas_id
2024-11-18 11:45:57 2024-11-18T10:45:57.637Z  INFO 1 --- [AAS Environment] [           main] o.e.d.b.a.base.DefaultAASEnvironment     : Uploading shell templateAas_id was successful!
2024-11-18 11:45:57 2024-11-18T10:45:57.761Z  WARN 1 --- [AAS Environment] [           main] ConfigServletWebServerApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'preconfigurationLoaderInitializer' defined in URL [jar:nested:/application/basyxExecutable.jar/!BOOT-INF/lib/basyx.aasenvironment-core-2.0.0-SNAPSHOT.jar!/org/eclipse/digitaltwin/basyx/aasenvironment/preconfiguration/PreconfigurationLoaderInitializer.class]: Unable to link the element with id 'testAas_id' with the Registry.
2024-11-18 11:45:57 2024-11-18T10:45:57.764Z  INFO 1 --- [AAS Environment] [           main] o.apache.catalina.core.StandardService   : Stopping service [Tomcat]
2024-11-18 11:45:57 2024-11-18T10:45:57.788Z  INFO 1 --- [AAS Environment] [           main] .s.b.a.l.ConditionEvaluationReportLogger : 
2024-11-18 11:45:57 
2024-11-18 11:45:57 Error starting ApplicationContext. To display the condition evaluation report re-run your application with 'debug' enabled.
2024-11-18 11:45:57 2024-11-18T10:45:57.808Z ERROR 1 --- [AAS Environment] [           main] o.s.boot.SpringApplication               : Application run failed
2024-11-18 11:45:57 
2024-11-18 11:45:57 org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'preconfigurationLoaderInitializer' defined in URL [jar:nested:/application/basyxExecutable.jar/!BOOT-INF/lib/basyx.aasenvironment-core-2.0.0-SNAPSHOT.jar!/org/eclipse/digitaltwin/basyx/aasenvironment/preconfiguration/PreconfigurationLoaderInitializer.class]: Unable to link the element with id 'testAas_id' with the Registry.
2024-11-18 11:45:57 at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1806) ~[spring-beans-6.1.14.jar!/:6.1.14]
2024-11-18 11:45:57 at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:600) ~[spring-beans-6.1.14.jar!/:6.1.14]
2024-11-18 11:45:57 at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:522) ~[spring-beans-6.1.14.jar!/:6.1.14]
2024-11-18 11:45:57 at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:337) ~[spring-beans-6.1.14.jar!/:6.1.14]
2024-11-18 11:45:57 at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-6.1.14.jar!/:6.1.14]
2024-11-18 11:45:57 at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:335) ~[spring-beans-6.1.14.jar!/:6.1.14]
2024-11-18 11:45:57 at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:200) ~[spring-beans-6.1.14.jar!/:6.1.14]
2024-11-18 11:45:57 at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:975) ~[spring-beans-6.1.14.jar!/:6.1.14]
2024-11-18 11:45:57 at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:971) ~[spring-context-6.1.14.jar!/:6.1.14]
2024-11-18 11:45:57 at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:625) ~[spring-context-6.1.14.jar!/:6.1.14]
2024-11-18 11:45:57 at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:146) ~[spring-boot-3.3.5.jar!/:3.3.5]
2024-11-18 11:45:57 at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:754) ~[spring-boot-3.3.5.jar!/:3.3.5]
2024-11-18 11:45:57 at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:456) ~[spring-boot-3.3.5.jar!/:3.3.5]
2024-11-18 11:45:57 at org.springframework.boot.SpringApplication.run(SpringApplication.java:335) ~[spring-boot-3.3.5.jar!/:3.3.5]
2024-11-18 11:45:57 at org.springframework.boot.SpringApplication.run(SpringApplication.java:1363) ~[spring-boot-3.3.5.jar!/:3.3.5]
2024-11-18 11:45:57 at org.springframework.boot.SpringApplication.run(SpringApplication.java:1352) ~[spring-boot-3.3.5.jar!/:3.3.5]
2024-11-18 11:45:57 at org.eclipse.digitaltwin.basyx.aasenvironment.component.AasEnvironmentComponent.main(AasEnvironmentComponent.java:42) ~[!/:2.0.0-SNAPSHOT]
2024-11-18 11:45:57 at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
2024-11-18 11:45:57 at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[na:na]
2024-11-18 11:45:57 at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
2024-11-18 11:45:57 at java.base/java.lang.reflect.Method.invoke(Method.java:569) ~[na:na]
2024-11-18 11:45:57 at org.springframework.boot.loader.launch.Launcher.launch(Launcher.java:102) ~[basyxExecutable.jar:2.0.0-SNAPSHOT]
2024-11-18 11:45:57 at org.springframework.boot.loader.launch.Launcher.launch(Launcher.java:64) ~[basyxExecutable.jar:2.0.0-SNAPSHOT]
2024-11-18 11:45:57 at org.springframework.boot.loader.launch.JarLauncher.main(JarLauncher.java:40) ~[basyxExecutable.jar:2.0.0-SNAPSHOT]
2024-11-18 11:45:57 Caused by: org.eclipse.digitaltwin.basyx.core.exceptions.RepositoryRegistryLinkException: Unable to link the element with id 'testAas_id' with the Registry.
2024-11-18 11:45:57 at org.eclipse.digitaltwin.basyx.aasrepository.feature.registry.integration.RegistryIntegrationAasRepository.registerAas(RegistryIntegrationAasRepository.java:146) ~[basyx.aasrepository-feature-registry-integration-2.0.0-SNAPSHOT.jar!/:2.0.0-SNAPSHOT]
2024-11-18 11:45:57 at org.eclipse.digitaltwin.basyx.aasrepository.feature.registry.integration.RegistryIntegrationAasRepository.createAas(RegistryIntegrationAasRepository.java:88) ~[basyx.aasrepository-feature-registry-integration-2.0.0-SNAPSHOT.jar!/:2.0.0-SNAPSHOT]
2024-11-18 11:45:57 at org.eclipse.digitaltwin.basyx.aasenvironment.environmentloader.IdentifiableUploader$DelegatingIdentifiableRepository.create(IdentifiableUploader.java:158) ~[basyx.aasenvironment-core-2.0.0-SNAPSHOT.jar!/:2.0.0-SNAPSHOT]
2024-11-18 11:45:57 at org.eclipse.digitaltwin.basyx.aasenvironment.environmentloader.IdentifiableUploader.create(IdentifiableUploader.java:94) ~[basyx.aasenvironment-core-2.0.0-SNAPSHOT.jar!/:2.0.0-SNAPSHOT]
2024-11-18 11:45:57 at org.eclipse.digitaltwin.basyx.aasenvironment.environmentloader.IdentifiableUploader.upload(IdentifiableUploader.java:55) ~[basyx.aasenvironment-core-2.0.0-SNAPSHOT.jar!/:2.0.0-SNAPSHOT]
2024-11-18 11:45:57 at org.eclipse.digitaltwin.basyx.aasenvironment.base.DefaultAASEnvironment.createShellsOnRepositoryFromEnvironment(DefaultAASEnvironment.java:215) ~[basyx.aasenvironment-core-2.0.0-SNAPSHOT.jar!/:2.0.0-SNAPSHOT]
2024-11-18 11:45:57 at org.eclipse.digitaltwin.basyx.aasenvironment.base.DefaultAASEnvironment.loadEnvironment(DefaultAASEnvironment.java:151) ~[basyx.aasenvironment-core-2.0.0-SNAPSHOT.jar!/:2.0.0-SNAPSHOT]
2024-11-18 11:45:57 at org.eclipse.digitaltwin.basyx.aasenvironment.preconfiguration.AasEnvironmentPreconfigurationLoader.loadPreconfiguredEnvironments(AasEnvironmentPreconfigurationLoader.java:91) ~[basyx.aasenvironment-core-2.0.0-SNAPSHOT.jar!/:2.0.0-SNAPSHOT]
2024-11-18 11:45:57 at org.eclipse.digitaltwin.basyx.aasenvironment.preconfiguration.PreconfigurationLoaderInitializer.loadPreconfiguredEnvironment(PreconfigurationLoaderInitializer.java:68) ~[basyx.aasenvironment-core-2.0.0-SNAPSHOT.jar!/:2.0.0-SNAPSHOT]
2024-11-18 11:45:57 at org.eclipse.digitaltwin.basyx.aasenvironment.preconfiguration.PreconfigurationLoaderInitializer.afterPropertiesSet(PreconfigurationLoaderInitializer.java:60) ~[basyx.aasenvironment-core-2.0.0-SNAPSHOT.jar!/:2.0.0-SNAPSHOT]
2024-11-18 11:45:57 at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1853) ~[spring-beans-6.1.14.jar!/:6.1.14]
2024-11-18 11:45:57 at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1802) ~[spring-beans-6.1.14.jar!/:6.1.14]
2024-11-18 11:45:57 ... 23 common frames omitted
2024-11-18 11:45:57 Caused by: org.eclipse.digitaltwin.basyx.aasregistry.client.ApiException: postAssetAdministrationShellDescriptor call failed with: 400 - {"messages":[{"code":"Bad Request","correlationId":null,"messageType":"Exception","text":"Field error in object 'assetAdministrationShellDescriptor' on field 'administration.templateId': rejected value [templateAas_id]; codes [Pattern.assetAdministrationShellDescriptor.administration.templateId,Pattern.administration.templateId,Pattern.templateId,Pattern.java.lang.String,Pattern]; arguments [org.springframework.context.support.DefaultMessageSourceResolvable: codes [assetAdministrationShellDescriptor.administration.templateId,administration.templateId]; arguments []; default message [administration.templateId],[Ljakarta.validation.constraints.Pattern$Flag;@3bce7351,^([\\t\\n\\r \\ud7ff\\ue000-\\ufffd]|\\ud800[\\udc00-\\udfff]|[\\ud801-\\udbfe][\\udc00-\\udfff]|\\udbff[\\udc00-\\udfff])*$]; default message [must match \"^([\\t\\n\\r \\ud7ff\\ue000-\\ufffd]|\\ud800[\\udc00-\\udfff]|[\\ud801-\\udbfe][\\udc00-\\udfff]|\\udbff[\\udc00-\\udfff])*$\"]","timestamp":1731926757.744419187}]}
2024-11-18 11:45:57 at org.eclipse.digitaltwin.basyx.aasregistry.client.api.RegistryAndDiscoveryInterfaceApi.getApiException(RegistryAndDiscoveryInterfaceApi.java:125) ~[basyx.aasregistry-client-native-2.0.0-SNAPSHOT.jar!/:2.0.0-SNAPSHOT]
2024-11-18 11:45:57 at org.eclipse.digitaltwin.basyx.aasregistry.client.api.RegistryAndDiscoveryInterfaceApi.postAssetAdministrationShellDescriptorWithHttpInfo(RegistryAndDiscoveryInterfaceApi.java:833) ~[basyx.aasregistry-client-native-2.0.0-SNAPSHOT.jar!/:2.0.0-SNAPSHOT]
2024-11-18 11:45:57 at org.eclipse.digitaltwin.basyx.aasregistry.client.api.RegistryAndDiscoveryInterfaceApi.postAssetAdministrationShellDescriptor(RegistryAndDiscoveryInterfaceApi.java:811) ~[basyx.aasregistry-client-native-2.0.0-SNAPSHOT.jar!/:2.0.0-SNAPSHOT]
2024-11-18 11:45:57 at org.eclipse.digitaltwin.basyx.aasrepository.feature.registry.integration.RegistryIntegrationAasRepository.registerAas(RegistryIntegrationAasRepository.java:142) ~[basyx.aasrepository-feature-registry-integration-2.0.0-SNAPSHOT.jar!/:2.0.0-SNAPSHOT]
2024-11-18 11:45:57 ... 34 common frames omitted
2024-11-18 11:45:57 

```

--- a/basyx.aasregistry/open-api/patch-base-extensions.yaml
+++ b/basyx.aasregistry/open-api/patch-base-extensions.yaml
@@ -92,7 +92,7 @@
   value: "^(0|[1-9][0-9]{0,3})$"
 - op: replace
   path: /components/schemas/AdministrativeInformation/allOf/1/properties/templateId/pattern
-  value: "^([\\t\\n\\r \\ud7ff\\ue000-\\ufffd]|\\ud800[\\udc00-\\udfff]|[\\ud801-\\udbfe][\\udc00-\\udfff]|\\udbff[\\udc00-\\udfff])*$"
+  value: "^([\\x09\\x0a\\x0d\\x20-\\ud7ff\\ue000-\\ufffd]|\\ud800[\\udc00-\\udfff]|[\\ud801-\\udbfe][\\udc00-\\udfff]|\\udbff[\\udc00-\\udfff])*$"
 ## EmbeddedDataSpecification
 - op: replace
   path: /components/schemas/EmbeddedDataSpecification/properties/dataSpecificationContent/$ref

